### ISSUE TITLE
Decide upon initial templating support

### ISSUE THREAD

**Issue by oubiwann** [lfe-dragon/dragon#4]
I lean heavily in favour of logic-less templates, so likely something mustache-based. What does rebar3 use for its templating? That team tends to make really excellent engineering decisions ...

**Comment by John-Goff** [comment]
Is making Elixir/Phoenix a requirement for Dragon a viable option? Seems like based on #1 and #2 that it would at least be an optional requirement. But if that's the case then I would push for [Phoenix.Component](https://hexdocs.pm/phoenix_live_view/Phoenix.Component.html) and HEEx templates. That way we can also integrate LFE templates with LiveView, since LiveView requires HEEx or something compatible with it to do change tracking.

**Comment by oubiwann** [comment]
At it's core, Dragon should be 100% LFE/Erlang. In other words, library-only. No need to pull in anything more complex (not counting client-level stuff like tailwind, sass, TS/JS compilation, etc.). The Erlang compiler should be all that's necessary (and LFE is just an Erlang lib).

There should be additional project that support integration with Elixir/Phoenix, but that should always be 100% optional.

A template solution for Dragon should be simple, easy-to-learn, and logic-less. I think erlydtl is too much for this. I've been meaning to see what rebar3 use ... I think it's some moustache variant?

**Comment by oubiwann** [comment]
Just read their template source code, and they use this:
* https://github.com/soranoba/bbmustache

**Comment by oubiwann** [comment]
Yeah, I love this. Super-simple. Keep the logic in code.

**Comment by lpil** [comment]
Hello! I imagined that in LFE one would use a HTML sexpr DSL, as that was what I'd seen in other Lisps. What do you prefer about templates in this case? Thanks!

**Comment by oubiwann** [comment]
@lpil yeah, I've used those before, too. In fact, I wrote one for LFE _ages_ ago ;-) 

While that's always a nice option, it's not the most convenient way to manage the design of a site -- and the larger the site, the more problematic it becomes to manage it with code instead of markup.

Lastly, it's really nice when you have a designer handy and you can just give them HTML and they can give back better HTML. This makes getting (sometimes excellent) deign help _much_ easier. Or even _possible_.
